### PR TITLE
Clean up Intent.{cpp,h}

### PR DESCRIPTION
**Pull request by Gymnasiast**
This gets rid of the C transfer functions, adds the licence and does some alignment fixes

### CODE REVIEW

File: src/openrct2-ui/input/KeyboardShortcut.cpp [hunk 1]
```diff
@@ -592,11 +592,11 @@ static void shortcut_quick_save_game()
     }
     else if (gScreenFlags & SCREEN_FLAGS_SCENARIO_EDITOR)
     {
-        Intent * intent = intent_create(WC_LOADSAVE);
-        intent_set_uint(intent, INTENT_EXTRA_LOADSAVE_TYPE, LOADSAVETYPE_SAVE | LOADSAVETYPE_LANDSCAPE);
-        intent_set_string(intent, INTENT_EXTRA_PATH, gS6Info.name);
+        Intent * intent = new Intent(WC_LOADSAVE);
```
Comment (AaronVanGeffen) [inline src/openrct2-ui/input/KeyboardShortcut.cpp]: You don't need to heap-allocate Intents using `new` if you're calling their destructor (`delete`) within the same function. Just keep them on the stack.

Comment (Gymnasiast) [reply src/openrct2-ui/input/KeyboardShortcut.cpp]: This is how it was done before. I only removed the wrappers.

Comment (marijnvdwerf) [reply src/openrct2-ui/input/KeyboardShortcut.cpp]: Also use `auto` for the type instead.

**Comment by Gymnasiast** [comment]
Ok, I made the changes, keeping `new`/`delete` only when calls to `create_save_game_as_intent()` were involved.

**Comment by AaronVanGeffen** [comment]
Looks a lot better. Thanks for making the changes.

`create_save_game_as_intent` and `window_save_prompt_mouseup` should probably have their `new`/`delete` approach refactored into using `std::unique_ptr`, too. Could be done in a separate PR, too, I suppose.

**Pull request merged**